W 11815997 rpa builder workbench cv (#84)
mulesoft/docs-rpa: rpa-builder/6.0/modules/ROOT/pages/managing-user-templates.adoc

> = Managing Templates

Reviewer: Added "User" to match what is in the nav.

> RPA Builder enables you to save Action Steps with all settings as templates that you can reuse in the project and you can also Import and Export templates.

Reviewer: This sentence isn't quite right because these are two separate ideas. Maybe:

RPA Builder enables you to save Action Steps and their settings as templates that you can reuse in the project. You can Import and Export these templates.

== Before you Begin

Consider the following when creating a template:

> * Search Pattern and Pattern Select Action Steps and all combinations of Action Steps that include them, cannot be saved as a template.

Reviewer: ```suggestion
* Search Pattern and Pattern Select Action Steps and all combinations of Action Steps that include them cannot be saved as a template.
```

* RPA Builder removes all passwords from templates before creating them.

== Create a Template

You can create a template using the _Context menu_ or the _Workflow Tools_.

* To create a Template from the _Context menu_:
+

> . Select the Action Step you want to save as a template.

Reviewer: Do you select and then right-click or is this really one step?

. Right-click it to open the context menu.
. Click *Save As Template*.

* To create a Template using the _Workflow Tools_:
+
. Select the Action Step you want to save as a template.
. In the Workbench Toolbar, click the *Tools* (image:wrench-icon.png[The Tools icon, 2%, 2%]) button.
. Click *Save Selected Items as Template*.

> After selecting any of these options, a *Save as* dialog opens in which you continue configuring the template:

Reviewer: The formatting in the preview is off here. See the List Continuation (aka Indented Paragraphs) in the Wiki for tips on lining things up: https://confluence.internal.salesforce.com/display/MTDT/Formatting+with+Asciidoc

[start=4]
. Select the path to the folder in which to store the template.
+
By default, the RPA Builder template folder is `\MuleSoft\RPA\UserTemplates`.

> . Enter a name for the template and click Save.

Reviewer: Should Save be bolded?

+
The Template details window that subsequently opens enables you to add additional information to this template.
+
image:save-as-template.png[The Template details window, 30%, 30%]
+

> When *Clear user data* is unselected, the template keeps the settings configured, so specific measurement Workflow elements, like starting MS Edge with a blank page, can be saved as a template as a whole and reused directly from the User Templates category in the Toolbox.

Reviewer: When you unselect *Clear user data*, the template...

> Create a set of templates before creating a Workflow to increase the creation speed significantly. Save the templates before configuring the Workflow objects, so the templates stay flexible in use.

Reviewer: ```suggestion
Creating a set of templates before creating a Workflow increases the creation speed significantly. Save the templates before configuring the Workflow objects, so the templates stay flexible.
```

* (image:delete-icon-alt.png[The Delete template button, 2%, 2%]) *Delete*
+
Deletes the selected Template.
* (image:import-icon.png[The Import template button, 2%, 2%]) *Import*
+
After you click *Import*, Windows Explorer opens and you can enter the path to the template. Select a template and the following window opens:
+
image:import-user-template.png[The Import template window, 50%, 50%]
+
You can verify the template again and change the name if necessary. Click *OK* to import the template into the Template Manager, where it becomes available for further use.
* (image:export-icon.png[The Export template button, 2%, 2%]) *Export*
+
When you click *Export* after selecting a template, the following window opens:
+
image:export-user-template.png[The Export template window, 50%, 50%]
+
In this window, you can change the name and navigate to the target folder where to save the exported template.

== Example Templates

The following template is a typical template for a normal Transaction with mouse activity:

image:template-transaction-mouse.png[A template showing a transaction with mouse activity, 50%, 50%]

> A useful variation of this template could be one that additionally uses keystrokes:

Reviewer: ```suggestion
A useful variation of this template could be one that uses keystrokes:
```

* (image:delete-icon-alt.png[The Delete template button, 2%, 2%]) *Delete*
+
Deletes the selected Template.
* (image:import-icon.png[The Import template button, 2%, 2%]) *Import*
+
After you click *Import*, Windows Explorer opens and you can enter the path to the template. Select a template and the following window opens:
+
image:import-user-template.png[The Import template window, 50%, 50%]
+
You can verify the template again and change the name if necessary. Click *OK* to import the template into the Template Manager, where it becomes available for further use.
* (image:export-icon.png[The Export template button, 2%, 2%]) *Export*
+
When you click *Export* after selecting a template, the following window opens:
+
image:export-user-template.png[The Export template window, 50%, 50%]
+
In this window, you can change the name and navigate to the target folder where to save the exported template.

== Example Templates

The following template is a typical template for a normal Transaction with mouse activity:

image:template-transaction-mouse.png[A template showing a transaction with mouse activity, 50%, 50%]

> A useful variation of this template could be one that uses keystrokes:

Reviewer 2: > A useful variation of this template could be one 

Sounds too verbose. Maybe something simplerL

> A useful variation of this template could use keystrokes.
